hx_lti_initializer/templates/hx_lti_initializer/dashboard_view.html

```diff
@@ -134,39 +134,47 @@ <h4 class="panel-title">
         $('.input-group #search_param').val(param);
         // Update search type
         type = param;
+        // Update view
+        update($('#studentsearch').val().toLowerCase());
+        
     });
     
-    // Get all panels 
-    var $panels = $('.panel');
-    
     // Search on key press
-    $('#studentsearch').on('keyup', function(){
+    $('#studentsearch').on('keyup', function(e){
         var val = this.value.toLowerCase();
         
-        if (type == 'users') {
-            // Show panels that match search value
+        // Only trigger if alphanumeric character is entered
+        if ((e.which <= 90 && e.which >= 48)) {
+            update(val);
+        }
+        
+        // Collapse and show all panels
+        else if (val == '') {
+            $panels.show();
+            $('.panel-collapse').collapse('hide');
+        }
+    });
+    
+    // Updates view
+    function update(value) {
+        if (type == 'name' && value != '') {
+            // Show panels that match user
             $panels.show().filter(function(){
                 $('.panel-collapse').collapse('show');
                 var title = $(this).find('.panel-title').text().toLowerCase();
-                return title.indexOf(val) < 0;
+                return title.indexOf(value) < 0;
             }).hide();
         }
         
-        else {
-            // Show panels that match search value
+        if (type == 'content' && value != '') {
+            // Show panels that match content
             $panels.show().filter(function(){
                 $('.panel-collapse').collapse('show');
                 var title = $(this).find('.panel-body').text().toLowerCase();
-                return title.indexOf(val) < 0;
+                return title.indexOf(value) < 0;
             }).hide();
```

Comment:
@arthurian Thanks for the suggestion. I thought that it was redundant when writing it myself, so will change.